src/lib/Behat/Component/UniversalDiscoveryWidget.php

```diff
@@ -194,6 +188,28 @@ public function selectInSearchResults(string $name): void
             ->click();
     }
 
+    public function clickCreateNewButton(): void
+    {
+        $this->getHTMLPage()->find($this->getLocator('createNewButton'))->click();
+    }
+
+    public function verifyCreateOnTheFlyFormIsLoaded(): void
+    {
+        $this->switchIntoUDWIframe('createIframe');
+        $this->getHTMLPage()->setTimeout(self::LONG_TIMEOUT)->find($this->getLocator('createOnTheFlyForm'))->assert()->isVisible();
+    }
+
+    public function switchIntoUDWIframe(string $iframeSelectorIdentifier): void
+    {
+        $iframeLocator = $this->getLocator($iframeSelectorIdentifier);
+        $script = sprintf("document.querySelector('%s').setAttribute('name','iframe')", $iframeLocator->getSelector());
+        $this->getHTMLPage()->setTimeout(self::SHORT_TIMEOUT)->waitUntilCondition(
+            new ElementExistsCondition($this->getHTMLPage(), $iframeLocator)
+        );
+        $this->getHTMLPage()->executeJavaScript($script);
+        $this->getSession()->switchToIFrame('iframe');
+    }
```

Comment:
```suggestion
    public function switchIntoUDWIframe(): void
    {
        $iframeLocator = $this->getLocator('iframe');
        $script = sprintf("document.querySelector('%s').setAttribute('name','iframe')", $iframeLocator->getSelector());
        $this->getHTMLPage()->setTimeout(self::SHORT_TIMEOUT)->waitUntilCondition(
            new ElementExistsCondition($this->getHTMLPage(), $iframeLocator)
        );
        $this->getHTMLPage()->executeJavaScript($script);
        $this->getSession()->switchToIFrame('iframe');
    }
```

This solution is based on a single selector matching both iframes in Content and Edit view